packages/editor/src/components/rich-text/index.js

```diff
@@ -440,6 +440,12 @@ export class RichText extends Component {
 			( keyCode === BACKSPACE && isHorizontalEdge( rootNode, true ) ) ||
 			( keyCode === DELETE && isHorizontalEdge( rootNode, false ) )
 		) {
+			// These modifiers would not result in modification of text content,
+			// if they're pressed they should not cause merge or deletion of blocks.
+			if ( ( event.metaKey || event.ctrlKey || event.altKey ) && event.shiftKey ) {
```

Comment:
I feel like these very localised rules about what are actually native keyboard behaviours are a bad idea, and lead to bugs like the one I found yesterday (#8639). The alternative is probably a pretty dramatic refactor though, so shouldn't be carried out on this PR.

If there is a better way, do let me know :)

Reply:
Not if I'm misunderstanding this, but when I have my cursor at the start of a paragraph block and press <kbd>Shift</kbd>+<kbd>Option</kbd>+<kbd>Backspace</kbd>, it still merges the blocks.

Reply:
@noisysocks Ugh, it does look like I made a few false assumptions. I've addressed them here:
e3f5a41e1c39e8c7045c03d8ee50d83216c4f937

The general logic is that the event shouldn't be handled if a modifier + delete/backspace wouldn't result in any text being deleted. Without the early return, some browser/native shortcuts are being swallowed by the RichText block when at the start or end of text content. The event for our delete block shortcut also wasn't being propagated in those cases.